src/figuresClasses.ts

```diff
@@ -1,11 +1,74 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
+    public trianglePerimeter: number = a + b + c,
+  ) {
+    const longestSide: number = Math.max(a, b, c);
+    const shortestSide: number = Math.min(a, b, c);
+    const otherSide: number = trianglePerimeter - longestSide - shortestSide;
```

Comment:
[CODE STYLE] - The variable `trianglePerimeter` is not defined in this scope. You should use `this.trianglePerimeter` to access the class property.